File: src/test/java/com.example.smarthome/SmartHomeTest.java
package com.example.smarthome;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class SmartHomeTest {

    @Test
    public void whenSwitchIsOn_turnsBulbOn() {
        Switch aSwitch = new AlwaysOnSwitch();
        BulbSpy bulbSpy = new BulbSpy();

        SmartHome smartHome = new SmartHome(aSwitch, bulbSpy);
        smartHome.run();

        assertEquals(true, bulbSpy.turnOnWasCalled);
    }

}

File: js/src/SmartHome/SmartHome.test.ts
import {Switch} from './Switch';
import {Switchable} from './Switchable';
import {AlwaysOnSwitch} from './AlwaysOnSwitch.test';
import {SmartHome} from './SmartHome';

describe('SmartHome', () => {
  let aSwitch: Switch;
  let bulbSpy: Switchable;
  let smartHome: SmartHome;

  beforeEach(() => {
    aSwitch = new AlwaysOnSwitch();
    bulbSpy = {
      turnOn: () => {},
      turnOff() {}
    };
    spyOn(bulbSpy, 'turnOn');

    smartHome = new SmartHome(aSwitch, bulbSpy);
    smartHome.run();
  });

  describe('when the switch is on', () => {
    it('turns the bulb on', () => {
      expect(bulbSpy.turnOn).toHaveBeenCalled();
    });
  });
});
